domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/frequencies/Frequency.java

```diff
@@ -0,0 +1,228 @@
+     * {@link Frequency}.
+     */
+    public static class FrequencyBuilder {
+        private String tripId;
+        private Integer startTime;
+        private Integer endTime;
+        private Integer headwaySecs;
+        private ExactTimes exactTimes;
+        private Integer originalExactTimes;
+        private final List<Notice> noticeCollection = new ArrayList<>();
+
+        /**
+         * Sets field tripId value and returns this
+         *
+         * @param tripId identifies a trip class
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder tripId(@NotNull String tripId) {
+            this.tripId = tripId;
+            return this;
+        }
+
+        /**
+         * Sets field startTime value and returns this
+         *
+         * @param startTime identifies a start time for a trip service at the first stop
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder startTime(@NotNull String startTime) {
+            this.startTime = convertHHMMSSToIntFromNoonOfDayOfService(startTime);
+            return this;
+        }
+
+        /**
+         * Sets field endTime value and returns this
+         *
+         * @param endTime identifies an end time for a trip service at the first stop
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder endTime(@NotNull String endTime) {
+            this.endTime = convertHHMMSSToIntFromNoonOfDayOfService(endTime);
+            return this;
+        }
+
+        /**
+         * Sets field endTime value and returns this
+         *
+         * @param headwaySecs identifies times in seconds between departures in the same headway for a trip service
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder headwaySecs(@NotNull Integer headwaySecs) {
+            this.headwaySecs = headwaySecs;
+            return this;
+        }
+
+        /**
+         * Sets field exactTimes value and returns this
+         *
+         * @param exactTimes identifies an end time for a trip
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder exactTimes(@Nullable Integer exactTimes) {
+            this.exactTimes = ExactTimes.fromInt(exactTimes);
+            this.originalExactTimes = exactTimes;
+            return this;
+        }
+
+        public EntityBuildResult<?> build() {
+            EntityBuildResult<?> toReturn;
+            if (tripId == null || startTime == null || endTime == null || headwaySecs == null ||
+                    headwaySecs < 0 || exactTimes == null) {
+
+                if (tripId == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "trip_id", tripId));
+                }
+
+                if (startTime == null) {
```

Comment:
My only concern would be if `MissingRequiredValueNotice` is always an accurate message here. I believe `startTime` could also be `null` if the conversion from HH:MM:SS to the int format fails? So the value may have been provided but not valid HH:MM:SS. But earlier in the type validation those types of errors should have been caught, right? So we know it's a valid HH:MM:SS value here?

Reply:
You're right, it should be an `InvalidTimeNotice`

Reply:
After discussion, we will implement the following strategy

- early fail on errors by default
- add an option to override that behavior, in which case it will be up to the person running the validator to understand what happened

As a consequence this code will stay unchanged

Reply:
#306